pyqg/qg_model.py

```diff
@@ -289,31 +289,39 @@ def _initialize_model_diagnostics(self):
             dims=('time',)
        )
 
+        # Obtain S matrix
+        def get_S_matrix(self):
+            try:
+                getattr(self, 'S')
+            except:
+                self.S = np.ma.zeros((self.nz, self.nz, self.nl, self.nk), np.dtype('float64'))
+                self.S[0,0] = -self.F1
+                self.S[0,1] = self.F1
+                self.S[1,0] = self.F2
+                self.S[1,1] = -self.F2
+        get_S_matrix(self)
```

Comment:
(if we want to avoid spuriously recomputing this matrix we can also do 

```python
from functools import cached_property

@cached_property
def S(self):
   # ...
```
)

Reply:
Good point... it's a bit unclean in my original code. I will create `self.S` in function `_initialize_inversion_matrix` in `qg_model.py` then.